### PR TITLE
Feature/create item

### PR DESCRIPTION
**Pull request by tpetersen0308**
POST requests to `/api/users/:user_uuid/items` create a new item record in the db that is associated with a user identified by `:user_uuid` URL parameter.

TODO:
- Add request authentication
- Add user authorization
- Seed the database with associated resources
- Validate posted item data

### CODE REVIEW

File: lib/rebay_api_web/controllers/auth_controller.exs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    signin(conn, changeset)
+  end
+
+  defp signin(conn, changeset) do
+    {:ok, user} = insert_or_update_user(changeset)
+    conn
+    |> put_session(:user_id, user.uuid)
+    |> redirect_to(to: user_path(conn, :show, user))
+    # |> redirect(to: UserController.show(conn, %{"id" => user.uuid}))
```
Comment (zacholauson) [inline lib/rebay_api_web/controllers/auth_controller.exs:25]: Remove this dead code?

File: lib/rebay_api_web/controllers/item_controller.ex [hunk 2]
```diff
@@ -1,43 +1,83 @@
...
+    item = Listings.get_item!(uuid)
 
     with {:ok, %Item{} = item} <- Listings.update_item(item, item_params) do
       render(conn, "show.json", item: item)
     end
   end
 
-  def delete(conn, %{"id" => id}) do
-    item = Listings.get_item!(id)
+  def delete(conn, %{"uuid" => uuid, "user_uuid" => _user_uuid}) do
```
Comment (zacholauson) [inline lib/rebay_api_web/controllers/item_controller.ex:45]: It doesn't look like there's anything here that would prevent me from modifying someone else's item, as long as I'm logged in... is that true?

Comment (tpetersen0308) [reply lib/rebay_api_web/controllers/item_controller.ex:45]: Yes, we actually just noticed this while testing frontend integration!

Comment (tpetersen0308) [reply lib/rebay_api_web/controllers/item_controller.ex:45]: This is resolved in PR #10

**Pull request merged**